### PR TITLE
Add gender

### LINKED ISSUE #49: Add a field Gender to Person
Update dependencies and tests
Update add command
Update edit command
Update User Guide

### PR DESCRIPTION
**Pull request by Jianwen0451**
Update dependencies and tests
Update add command
Update edit command
Update User Guide
close #49

### CODE REVIEW

**Review by RadieonAjax — APPROVED** [review]
Looks good to me! But some suggestions for the structure of the PR:

- Try to split the PR into smaller sections for easier review (for example):
    - Add support for Gender
    - Add test cases for Gender

Nonetheless, great job!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -76,15 +76,15 @@ Format: `help`
 
 Adds a person to the address book.
 
-Format: `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]…​`
+Format: `add n/NAME p/PHONE_NUMBER e/EMAIL g/gender a/ADDRESS [t/TAG]…​`
```
Comment (tanyqcollin) [inline docs/UserGuide.md:79]: gender should be uppercase to make it more consistent

File: docs/UserGuide.md [hunk 2]
```diff
@@ -96,7 +96,7 @@ Format: `list`
 
 Edits an existing person in the address book.
 
-Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`
+Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [g/gender][a/ADDRESS] [t/TAG]…​`
```
Comment (tanyqcollin) [inline docs/UserGuide.md:99]: missing space between `[g/gender]` and `[a/ADDRESS]` and the gender should be uppercase

File: src/main/java/seedu/address/model/person/Gender.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Student's gender in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGender(String)}
+ */
+public class Gender {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Gender should only be male or female. And should not be empty";
```
Comment (tanyqcollin) [inline src/main/java/seedu/address/model/person/Gender.java:13]: I think the sentence is a bit weird, you can change the message to this `"Gender should be either 'male' or 'female', and must not be empty."`

**Review by tanyqcollin — COMMENTED** [review]
overall is good but got some minor things to change

File: docs/UserGuide.md (hunk 1 above)
Comment (Jianwen0451) [reply docs/UserGuide.md]: OK change this part in the recent commit

File: docs/UserGuide.md (hunk 2 above)
Comment (Jianwen0451) [reply docs/UserGuide.md]: Ok change this part in the recent commit

File: src/main/java/seedu/address/model/person/Gender.java (hunk 3 above)
Comment (Jianwen0451) [reply src/main/java/seedu/address/model/person/Gender.java]: I changed the message a bit in the recent commit. It is more consistent with other message now

**Pull request merged**